packages/protocol/contracts/tokenvault/ERC1155Vault.sol

```diff
@@ -251,11 +263,14 @@ contract ERC1155Vault is BaseNFTVault, ERC1155ReceiverUpgradeable {
         )
     {
         return abi.decode(
-            _extractCalldata(dataWithSelector),
+            dataWithSelector[4:],
```

Comment:
I think you can even remove this function, and simply use the following in other place(s):
```
(, CanonicalNFT memory nft,
            address owner,
            address to,
            uint256[] memory tokenIds,
            uint256[] memory amounts) = abi.decode(bytes4, anonicalNFT, address, address, uint256[], uint256[])

```